### PR TITLE
Slight refractor to borg reboot boards, allows them to be reusable

### PR DESCRIPTION
**Pull request by zxaber**
## About The Pull Request
Refractors the cyborg reboot module to be an independent item rather than a subtype of cyborg upgrades. The new item is no longer consumed during use.

Rebooting now makes a different sound when rebooting an AI shell or ghostless borg, rather than no sound at all.

## Why It's Good For The Game
A recent PR fixed a longstanding apparent bug that allowed reboot boards to be resused. A bug it may have been, but the reboot board being a single-use item is kinda terrible, and doesn't make a lot of sense that the board is somehow destroyed when used. Mothblocks, who merged the bugfix PR, has given me permission to PR this change.

## Changelog
:cl:
balance: The cyborg emergency reboot module is now reusable.
sound: The cyborg emergency reboot module will now make a different sound when rebooting AI shells or borgs without a ghost.
/:cl:

### CODE REVIEW

**Comment by Colovorat** [comment]
i really didn't wanted to fix that, but grinding of gbp, you know

File: code/modules/vehicles/mecha/mech_fabricator.dm [hunk 1]
```diff
@@ -142,6 +142,9 @@
 					sub_category += "Engineering"
 			else
 				sub_category += "All Cyborgs"
+		else if(built_item in typesof(/obj/item/borg_restart_board)) //This is shitty but it wasn't I who made this code this way
```
Comment (Watermelon914) [inline code/modules/vehicles/mecha/mech_fabricator.dm:145]:
```suggestion
		else if(ispath(built_item, /obj/item/borg_restart_board))
```

**Comment by Fikou** [comment]
> A bug it may have been, but the reboot board being a single-use item is kinda terrible

why, it shouldnt be an item you print roundstart and then have it permanently as a tool

**Comment by Time-Green** [comment]
> > A bug it may have been, but the reboot board being a single-use item is kinda terrible
> 
> why, it shouldnt be an item you print roundstart and then have it permanently as a tool

Why not? Unless borg obesity is added, it wouldn't make sense either

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 2]
```diff
@@ -775,3 +755,35 @@
 	var/obj/item/pushbroom/cyborg/BR = locate() in R.model.modules
 	if (BR)
 		R.model.remove_module(BR, TRUE)
+
+///This isn't an upgrade or part of the same path, but I'm gonna just stick it here because it's a tool used on cyborgs.
+/obj/item/borg_restart_board
+	name = "cyborg emergency reboot module"
+	desc = "Used to force a reboot of a disabled-but-repaired cyborg, bringing it back online."
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/robot/robot_upgrades.dm:762]: can you put something in here that clarifies its reusable unlike the other boards

**Comment by ExcessiveUseOfCobblestone** [comment]
Why did you repath instead of just turning it off one use?

**Comment by zxaber** [comment]
Because that makes the board stay in the borg (like a thruster upgrade) and drop if the borg is deconstructed.

Previously, there was a very longstanding bug where using the upgrade board would have it drop to the ground and be re-usable. However, it spat out upgrade errors. A PR was merged that fixed this bug, and turned *on* the one_use var. I don't want to revert to the bug version, I want the reusable aspect to be intentionally in place.

This reboot module doesn't behave like an upgrade. It's not consumed or put into the borg like upgrades are. It just shares the same sprite and file.

**Comment by ExcessiveUseOfCobblestone** [comment]
So dumb lol ok pretend I merged this

**Comment by Fikou** [comment]
hi cobby

**Comment by ExcessiveUseOfCobblestone** [comment]
GMers but to the tune of zombies on my lawn where the flower says “screen door shield”

**Pull request merged**